pkg/ipamd/ipamd.go

```diff
@@ -2107,6 +2114,10 @@ func (c *IPAMContext) hasRoomForEni() bool {
 	if c.enablePodENI && c.dataStore.GetTrunkENI() == "" {
 		trunkEni = 1
 	}
+	if c.numNetworkCards > 1 {
+		// ENIs on non-zero network cards do not count
+		return c.dataStore.GetENIs() < (c.maxENI - c.unmanagedENI - trunkEni + c.nonZeroCardENI)
```

Comment:
I am a bit confused by this logic. Since we have more than 1 network card, I can see why we want to use `c.numNetworkCards` in this equation.

So to check if there is room, we start by getting how many ENIs are attached. Then we check if that is less than the max number of ENIs the instance can support, subtracting the number of ENIs that are unmanaged by IPAM (including trunk ENI). How does adding in the number of network cards help in this comparison? Is this so that we can attach an ENI to an additional network card in the future?

Reply:
Discussed offline, and we can change this equation to:
```
// For multi-card instances, c.unmanagedENI includes all ENIs attached to non-zero network cards. We must exclude these ENIs from the calculation as they cannot be used.
return c.dataStore.GetENIs() < (c.maxENI - (c.unmanagedENI - c.multiCardENIs) - trunkEni)
```
for more readability